src/alexandria3k/data_sources/issn_subject_codes.py

```diff
@@ -0,0 +1,198 @@
+            try:
+                serial_search = SerialSearch(query=query, view="STANDARD")
+                results = list(serial_search.results)
+
+                for result in results:
+                    for code in (result.get("subject_area_codes")).split(";"):
+                        writer.writerow([issn, code])
+
+            except (KeyError, ValueError):
+                warn(f"Error processing ISSN {issn}")
+
+    def download(self, database, data_location, sql_query=None):
+        """
+        Create a CSV file with ISSNs and their corresponding ASJC subject codes from API calls.
+
+        :param database: The path specifying the SQLite database to use in order to fetch the ISSNs.
+        :type database: str
+
+        :param data_location: The path specifying the CSV file to use for the population.
+        :type data_location: str
+
+        :param sql_query: The SQL query to use in order to fetch the ISSNs, defaults to `None`.
+            The default query fetches all unique ISSNs from the database. The query should
+            return a single column with the name `issn`.
+        :type sql_query: str, optional
+        """
+        try:
+            pybliometrics.scopus.init()
+        except RuntimeError as e:
+            raise Alexandria3kError(
+                "Error in downloading data with pybliometrics"
+            ) from e
+        connection = sqlite3.connect(database)
+        ensure_table_exists(connection, "works")
+        cursor = connection.cursor()
+        script = sql_query or get_string_resource("sql/get-issns.sql")
+        issns = self.execute_sql_query(cursor, script)
+
+        with open(
+            data_location, mode="w", newline="", encoding="utf-8"
+        ) as file:
+            writer = csv.writer(file)
+            writer.writerow(["issn", "subject_code"])
+            self.fetch_subject_codes(writer, issns)
+
+        connection.close()
+        perf.log("create_csv_from_api")
+
+    def populate(
+        self, database_path, columns=None, condition=None, data_location=None
+    ):
+        """
+        Populate the SQLite database with data from the CSV file.
+
+        :param database_path: The path specifying the SQLite database to populate.
+        :type database_path: str
+
+        :param columns: A list of strings specifying the columns to populate, defaults to `None`.
+        :type columns: list, optional
+
+        :param condition: SQL expression specifying the rows to include in the database's
+        population, defaults to `None`.
+        :type condition: str, optional
+
+        :param data_location: The path specifying the CSV file to use for the population,
+        defaults to `None`.
+        :type data_location: str, optional
+        """
+        # Update the data source to use the CSV file
+        new_data_source = VTSource(
+            issn_subject_codes_table, data_location, self.sample
+        )
+        self.data_source = new_data_source
```

Comment:
I think you can handle the above lines in `__init__` and dispense with the `populate` method entirely.

Reply:
I agree. However, i left it for the `data_location` parameter, in case the user wants to modify the parameter from `__init__`.

Reply:
The CLI should also be able to work with the constructor API, but passing it the appropriate arguments for each subcommand, no?

Reply:
I had a typo in my comment: "but passing" → "by passing".

Both populate and download should work.

Please try a small example through the CLI to make sure.

Reply:
I now removed the populate method from `IssnSubjectCodes`. I am testing now some cases from the CLI.

Reply:
You were correct (very unexpected, i know haha) ! I tested it and it works just as you said. I also added some necessary arguments that i had forgot.